### PR TITLE
#793 added FirstOf

### PR DESCRIPTION
**Pull request by krzyk**
#793 
- added `FirstOf` (as defined in the issue) and tests covering the functionality

### CODE REVIEW

File: src/test/java/org/cactoos/scalar/FirstOfTest.java [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+import org.cactoos.iterable.IterableOf;
+import org.cactoos.iterable.IterableOfInts;
+import org.cactoos.matchers.ScalarHasValue;
+import org.hamcrest.MatcherAssert;
+import org.junit.Test;
+
+/**
+ * Tests for {@link FirstOf}.
+ *
+ * <p>This class is thread-safe.
```
Comment (SharplEr) [inline src/test/java/org/cactoos/scalar/FirstOfTest.java]: @krzyk Remove this line.

**Comment by krzyk** [comment]
@SharplEr thanks for the review, here's an updated code

**Comment by SharplEr** [comment]
@llorllale Good to merge

**Comment by llorllale** [comment]
@krzyk #794 was merged into `master`. Please update your branch and make sure the build runs

**Comment by krzyk** [comment]
@llorllale updated

**Comment by llorllale** [comment]
@rultor merge

**Comment by 0crat** [comment]
@elenavolokhova[/z](https://www.0crat.com/u/elenavolokhova) please review this job completed by @sharpler[/z](https://www.0crat.com/u/sharpler), as in [§30](http://www.zerocracy.com/policy.html#30); the job will be fully closed and all payments will be made when the quality review is completed

**Comment by 0crat** [comment]
The job [#840](https://github.com/yegor256/cactoos/issues/840) is now out of scope

**Comment by elenavolokhova** [comment]
@0crat quality good

**Comment by 0crat** [comment]
Order was finished, quality is "good": +20 point(s) just awarded to @sharpler[/z](https://www.0crat.com/u/sharpler)

**Comment by 0crat** [comment]
Quality review completed: +8 point(s) just awarded to @elenavolokhova[/z](https://www.0crat.com/u/elenavolokhova)

**Pull request merged**